core/primitives/src/trie_key.rs

```diff
@@ -95,9 +96,45 @@ pub mod col {
         (BANDWIDTH_SCHEDULER_STATE, "BandwidthSchedulerState"),
         (BUFFERED_RECEIPT_GROUPS_QUEUE_DATA, "BufferedReceiptGroupsQueueData"),
         (BUFFERED_RECEIPT_GROUPS_QUEUE_ITEM, "BufferedReceiptGroupsQueueItem"),
+        (GLOBAL_CONTRACT_CODE, "GlobalContractCode"),
     ];
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, ProtocolSchema)]
+pub enum GlobalContractCodeIdentifier {
+    AccountId(AccountId),
+    CodeHash(CryptoHash),
+}
+
+impl GlobalContractCodeIdentifier {
+    pub fn len(&self) -> usize {
+        self.discriminator().len()
+            + match self {
+                Self::AccountId(account_id) => account_id.len(),
+                Self::CodeHash(hash) => hash.as_bytes().len(),
+            }
+    }
+
+    pub fn append_into(&self, buf: &mut Vec<u8>) {
```

Comment:
Why not use `borsh::to_vec().unwrap()` here instead of defining functions like append_into, discriminator, etc.?

Reply:
`TrieKey::AccessKey` has a similar pattern

Reply:
good point! using borsh is a bit less space-efficient because it has to include string length for account id, but that doesn't really matter for global contracts
updated and added tests in e21133be5